src/Hangfire.Mongo/MongoWriteOnlyTransaction.cs

```diff
@@ -146,11 +146,17 @@ public override void AddToSet(string key, string value)
 
         public override void AddToSet(string key, string value, double score)
         {
+            var builder = Builders<SetDto>.Update;
+            var set = builder.Set(_ => _.Score, score);
+            var setTypesOnInsert = builder.SetOnInsert("_t", new[] {nameof(KeyValueDto), nameof(ExpiringKeyValueDto), nameof(SetDto)});
+            var setExpireAt = builder.SetOnInsert(nameof(SetDto.ExpireAt), BsonNull.Value);
```

Comment:
Why not use 
`.SetOnInsert(_ => _.ExpireAt, BsonNull.Value)`
as you have access to the hard type?